lib/ex_aws/cloudwatch.ex

```diff
@@ -626,7 +626,9 @@ defmodule ExAws.Cloudwatch do
   end
 
   defp format_param({:metric_data, metric_data}) do
-    metric_data |> format(prefix: "MetricData.member")
+    metric_data
+    |> Enum.flat_map(&format_param/1)
```

Comment:
It seems not correct implementation to me...

Type spec of `metric_data` is `[metric_datum, ...]`, therefor the value passed to mapping function (`&format_param/1`) is like `[dimensions: [my_dim_1: "MyDimValue1", my_dim_2: "MyDimValue2"], metric_name: "MyTestMetic", timestamp: #DateTime<2018-06-20 02:51:43Z>, unit: "Count", value: 2]`

`Enum.flat_map(&format_param/1)` causes the error like this:

```
     ** (FunctionClauseError) no function clause matching in ExAws.Cloudwatch.format_param/1

     The following arguments were given to ExAws.Cloudwatch.format_param/1:
     
         # 1
         [dimensions: [my_dim_1: "MyDimValue1", my_dim_2: "MyDimValue2"], metric_name: "MyTestMetic", timestamp: #DateTime<2018-06-20 02:51:43Z>, unit: "Count", value: 2]
     
     Attempted function clauses (showing 10 out of 12):
     
         defp format_param({:alarm_actions, alarm_actions})
         defp format_param({:alarm_names, alarm_names})
         defp format_param({:dashboard_names, dashboard_names})
         defp format_param({:dimensions, dimensions})
         defp format_param({:end_time, end_time})
         defp format_param({:extended_statistics, extended_statistics})
         defp format_param({:insufficient_data_actions, insufficient_data_actions})
         defp format_param({:metric_data, metric_data})
         defp format_param({:ok_actions, ok_actions})
         defp format_param({:start_time, start_time})
```

Reply:
@yumeto 

How did you get to that error? The below is my implementation that is working with the PR I posted above. 

```
    [
      dimensions: [{:jobs_count, "Jobs"}],
      metric_name: "JobsCount",
      unit: "Count",
      value: PgJobRunner.count_jobs()
    ]
    |> ExAws.Cloudwatch.put_metric_data(namespace)
    |> ExAws.request()
```

This is posting up to CloudWatch without any errors from my end using the branch referenced in the PR.

Reply:
@CaldwellYSR I've pass the list of metric data

```
  [
     [
      dimensions: [{:jobs_count, "Jobs"}],
      metric_name: "JobsCount",
      unit: "Count",
      value: PgJobRunner.count_jobs()
      ],
     [
      dimensions: [{:jobs_count, "Jobs"}],
      metric_name: "OtherMetric",
      unit: "Count",
      value: 1
      ]
    ]
    |> ExAws.Cloudwatch.put_metric_data(namespace)
    |> ExAws.request()
```